### DISCUSSION TITLE
is `Untyped` necessary?

### DISCUSSION THREAD

**Discussion by DetachHead** [KotlinIsland/basedmypy#417] [Polls]
`Untyped` is just a type alias for `Any`. it has the exact same functionality and the only difference is the wording of the error messages.

imo this adds unnessecary confusion to users who see these two different types. i think most people would assume they have different functionality.

users already have to understand the difference between `object` and `Any`, i dont think theres much of a benefit to adding a third one if the only difference is error message wording.

although you could argue that they have slightly different usages, the same can also be said to argue for a type alias of `bool` to differentiate between bools that mean "enabled/disabled" and bools that mean "yes/no"

**Comment by KotlinIsland** [comment]
One of the main advantages of `Untyped` is in dev experience when seeing an error message, to quickly be able to see that something is `Untyped` is very informative as to what's going on, seeing an `Any` could imply various different things (an error, something is unable to be typed, etc).

There is also a good usecase for denoting incomplete stubs

We should do some user testing on this.
